cattle/plugins/docker/compute.py

```diff
@@ -763,6 +765,34 @@ def _setup_hostname(self, create_config, instance):
         except (KeyError, AttributeError):
             pass
 
+    def _setup_device_options(self, config, instance):
+        option_configs = \
+            [('readIops', [], 'BlkioDeviceReadIOps', 'Rate'),
+             ('writeIops', [], 'BlkioDeviceWriteIOps', 'Rate'),
+             ('readBps', [], 'BlkioDeviceReadBps', 'Rate'),
+             ('writeBps', [], 'BlkioDeviceWriteBps', 'Rate'),
+             ('weight', [], 'BlkioWeightDevice', 'Weight')]
+
+        try:
+            device_options = instance.data['fields']['deviceOptions']
+            for dev, options in device_options.iteritems():
+                if dev == 'DEFAULT_DISK':
+                    dev = self.host_info.get_default_disk()
+                    if not dev:
+                        log.warn("Couldn't find default device. Not setting"
+                                 "device options: %s", options)
+                        continue
+                for k, dev_list, _, field in option_configs:
+                    if k in options and options[k] is not None:
+                        value = options[k]
+                        dev_list.append({'Path': dev, field: value})
+
+            for _, dev_list, docker_field, _ in option_configs:
+                if len(dev_list):
+                    config[docker_field] = dev_list
+        except (KeyError, AttributeError):
+            pass
```

Comment:
These blocks that catch and silently swallow KeyError and AttributeErrors are the established pattern for handling backwards compatibility with cattle. The idea is that if this runs against a version of cattle that doesn't have the deviceOptions field, it will just silently move on.

But looking at this code, the try/except block has too much code in it. I could potentially mask legit errors.

What if I refactor so that it was basically this?
```
try:
    device_options = instance.data['fields']['deviceOptions']
except (KeyError, AttributeError):
    return
# continue with logic and let any exceptions just bubble up
```